parlai/agents/test_agents/counter.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Test agent which counts its number of unique items.
+"""
+
+from __future__ import annotations
+from typing import Tuple
+from collections import Counter
+
+from parlai.core.torch_agent import TorchAgent
+from parlai.core.metrics import Metric, SumMetric
+from parlai.core.message import Message
+
+
+class _CounterMetric(Metric):
+    __slots__ = ('_counter',)
+
+    def __init__(self, counter: Counter):
+        self._counter = counter
+
+    def __add__(self, other: Metric):
+        if other is None:
+            return self
+        counter = self._counter + other._counter
+        return type(self)(counter)
+
+
+class TimesSeenMetric(_CounterMetric):
+    """
+    Max number of times any example was seen.
+    """
+
+    def value(self) -> int:
+        if not self._counter:
+            return 0
+        return max(self._counter.values())
+
+
+class UniqueMetric(_CounterMetric):
+    """
+    Number of unique utterances.
+    """
+
+    def value(self) -> int:
+        if not self._counter:
+            return 0
+        return len(self._counter)
+
+
+class CounterAgent(TorchAgent):
+    """
+    Simple agent that counts the number of unique things it has seen.
+
+    Could be simpler, but we make it a TorchAgent so it's happy with dynamic batching.
+    """
+
+    def __init__(self, opt, shared=None):
+        self.model = self.build_model()
+        self.criterion = None
+        super().__init__(opt, shared)
+        self._counter: Counter[Tuple[str, str]]
+        if shared is None:
+            self._counter = Counter()
+            self._padding_counter = Counter()
+        else:
+            self._counter = shared['counter']
+            self._padding_counter = shared['padding']
+
+    def share(self):
+        shared = super().share()
+        shared['counter'] = self._counter
+        shared['padding'] = self._padding_counter
+        return shared
+
+    def save(self, path=None):
+        pass
+
+    def load(self, path=None):
+        pass
+
+    def _val(self, val):
+        if isinstance(val, list):
```

Comment:
this is a bit weird so maybe add a comment here why you do this (text vs. labels)